### PR TITLE
Implementation of the sensor VEML7700

### PR DESCRIPTION
**Pull request by Dr-Gandalf**
This code, implements the measurement of light (lux and irradiance), using the veml7700 light sensor. This pr is related to the pr [#498](https://github.com/meshtastic/protobufs/pull/498) on the protobufs repo 

It also modifies the MQTT client to publish the new signals to the mqtt server.

### CODE REVIEW

**Review by caveman99 — APPROVED** [review]

File: src/modules/Telemetry/EnvironmentTelemetry.cpp [hunk 1]
```diff
@@ -235,49 +238,68 @@ bool EnvironmentTelemetryModule::handleReceivedProtobuf(const meshtastic_MeshPac
...
+
 
     if (valid) {
-        LOG_INFO("(Sending): barometric_pressure=%f, current=%f, gas_resistance=%f, relative_humidity=%f, temperature=%f\n",
+        LOG_INFO("(Sending): barometric_pressure=%f, current=%f, gas_resistance=%f, relative_humidity=%f, temperature=%f, "
+                 "lux=%f\n",
                  m.variant.environment_metrics.barometric_pressure, m.variant.environment_metrics.current,
                  m.variant.environment_metrics.gas_resistance, m.variant.environment_metrics.relative_humidity,
-                 m.variant.environment_metrics.temperature);
+                 m.variant.environment_metrics.temperature, m.variant.environment_metrics.lux);
```
Comment (caveman99) [inline src/modules/Telemetry/EnvironmentTelemetry.cpp:302]: same here, please use the second line.

File: src/mqtt/MQTT.cpp [hunk 2]
```diff
@@ -671,6 +671,9 @@ std::string MQTT::meshPacketToJson(meshtastic_MeshPacket *mp)
                     msgPayload["gas_resistance"] = new JSONValue(decoded->variant.environment_metrics.gas_resistance);
                     msgPayload["voltage"] = new JSONValue(decoded->variant.environment_metrics.voltage);
                     msgPayload["current"] = new JSONValue(decoded->variant.environment_metrics.current);
+                    msgPayload["lux"] = new JSONValue(decoded->variant.environment_metrics.lux);
+                    msgPayload["white_lux"] = new JSONValue(decoded->variant.environment_metrics.white_lux);
+                    msgPayload["iaq"] = new JSONValue((uint)decoded->variant.environment_metrics.iaq);
```
Comment (caveman99) [inline src/mqtt/MQTT.cpp:676]: thanks! :-)

**Review by caveman99 — CHANGES_REQUESTED** [review]

File: src/modules/Telemetry/EnvironmentTelemetry.cpp [hunk 3]
```diff
@@ -214,13 +218,12 @@ bool EnvironmentTelemetryModule::handleReceivedProtobuf(const meshtastic_MeshPac
         const char *sender = getSenderShortName(mp);
 
         LOG_INFO("(Received from %s): barometric_pressure=%f, current=%f, gas_resistance=%f, relative_humidity=%f, "
-                 "temperature=%f\n",
+                 "temperature=%f, lux=%f\n",
                  sender, t->variant.environment_metrics.barometric_pressure, t->variant.environment_metrics.current,
                  t->variant.environment_metrics.gas_resistance, t->variant.environment_metrics.relative_humidity,
-                 t->variant.environment_metrics.temperature);
+                 t->variant.environment_metrics.temperature, t->variant.environment_metrics.lux);
```
[the remark this answers is not in the document]
Comment (Dr-Gandalf) [reply src/modules/Telemetry/EnvironmentTelemetry.cpp]: done

File: src/modules/Telemetry/EnvironmentTelemetry.cpp (hunk 1 above)
Comment (Dr-Gandalf) [reply src/modules/Telemetry/EnvironmentTelemetry.cpp]: done

File: src/mqtt/MQTT.cpp (hunk 2 above)
Comment (Dr-Gandalf) [reply src/mqtt/MQTT.cpp:676]: ;-)

**Review by thebentern — APPROVED** [review]

**Comment by thebentern** [comment]
Looks good to me. Upstream protobufs have conflicts now though

**Review by caveman99 — APPROVED** [review]

**Comment by caveman99** [comment]
@thebentern now you know why i tell people every time not to push the protobuf subdir or generated files. That always explodes

**Comment by thebentern** [comment]
> @thebentern now you know why i tell people every time not to push the protobuf subdir or generated files. That always explodes

image

**Pull request merged**